x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_form_advanced_options.test.tsx

```diff
@@ -0,0 +1,222 @@
+    expect(screen.getByText('Override')).toBeInTheDocument();
+    expect(screen.getByTestId('lookBackWindowRangeInput')).toHaveValue('6');
+    expect(screen.getByTestId('statusChangeThresholdRangeInput')).toHaveValue('4');
+    expect(screen.getByTestId('ruleSettingsFlappingMessage')).toHaveTextContent(
+      'An alert is flapping if it changes status at least 4 times in the last 6 rule runs.'
+    );
+
+    userEvent.click(screen.getByTestId('ruleFormAdvancedOptionsOverrideSwitch'));
+    expect(mockOnflappingChange).toHaveBeenCalledWith(null);
+  });
+
+  test('should not allow override when global flapping is off', async () => {
+    http.get.mockResolvedValue({
+      look_back_window: 10,
+      status_change_threshold: 3,
+      enabled: false,
+    });
+    useKibanaMock().services.http = http;
+
+    render(
+      <IntlProvider locale="en">
+        <QueryClientProvider client={queryClient}>
+          <RuleFormAdvancedOptions
+            enabledFlapping
+            alertDelay={5}
+            flappingSettings={{
+              lookBackWindow: 6,
+              statusChangeThreshold: 4,
+            }}
+            onAlertDelayChange={mockAlertDelayChange}
+            onFlappingChange={mockOnflappingChange}
+          />
+        </QueryClientProvider>
+      </IntlProvider>
+    );
+
+    expect(await screen.findByText('OFF')).toBeInTheDocument();
+    expect(screen.queryByText('Override')).not.toBeInTheDocument();
+    expect(screen.queryByTestId('ruleFormAdvancedOptionsOverrideSwitch')).not.toBeInTheDocument();
+    expect(screen.queryByTestId('ruleSettingsFlappingMessage')).not.toBeInTheDocument();
+  });
+
+  test('should allow for flapping inputs to be modified', async () => {
+    render(
+      <IntlProvider locale="en">
+        <QueryClientProvider client={queryClient}>
+          <RuleFormAdvancedOptions
+            enabledFlapping
+            alertDelay={5}
+            flappingSettings={{
+              lookBackWindow: 10,
+              statusChangeThreshold: 10,
+            }}
+            onAlertDelayChange={mockAlertDelayChange}
+            onFlappingChange={mockOnflappingChange}
+          />
+        </QueryClientProvider>
+      </IntlProvider>
+    );
+
+    expect(await screen.findByTestId('lookBackWindowRangeInput')).toBeInTheDocument();
+
+    const lookBackWindowInput = screen.getByTestId('lookBackWindowRangeInput');
+    const statusChangeThresholdInput = screen.getByTestId('statusChangeThresholdRangeInput');
+
+    // Change lookBackWindow to a smaller value
+    fireEvent.change(lookBackWindowInput, { target: { value: 5 } });
+    // statusChangeThresholdInput gets pinned to be 5
+    expect(mockOnflappingChange).toHaveBeenLastCalledWith({
+      lookBackWindow: 5,
+      statusChangeThreshold: 5,
+    });
+
+    // Try making statusChangeThreshold bigger
+    fireEvent.change(statusChangeThresholdInput, { target: { value: 20 } });
```

Comment:
I'm a bit confused here, shouldn't the values still be 5?

there was a change done before in line 180

```
expect(mockOnflappingChange).toHaveBeenLastCalledWith({
    lookBackWindow: 5,
    statusChangeThreshold: 5,
});
```

Anyway, I think we should have here different unit tests where we test the interaction with the sliders. Something like:

```
describe('should allow for flapping inputs to be modified', () => {
    test('change lookBackWindow to a smaller value', async () => {
        // confirm the threshold is also changed to the new value
    });

    test('change lookBackWindow to a smaller value but bigger to the threshold', async () => {
        // confirm the threshold remains unchanged
    });

    test('try changing the threshold to a value>lookBackWindow', async () => {
        // nothing happens; not allowed
    });

    test('try changing the threshold to a value<=lookBackWindow', async () => {
        // works
    });
});
```